components/playlist/browser/playlist_media_file_downloader.h

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PLAYLIST_BROWSER_PLAYLIST_MEDIA_FILE_DOWNLOADER_H_
+#define BRAVE_COMPONENTS_PLAYLIST_BROWSER_PLAYLIST_MEDIA_FILE_DOWNLOADER_H_
+
+#include <list>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/files/file_path.h"
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+#include "base/values.h"
+
+namespace base {
+class FilePath;
+class SequencedTaskRunner;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+class GURL;
+
+namespace playlist {
+
+// Handle one Playlist at once.
+class PlaylistMediaFileDownloader {
+ public:
+  class Client {
+   public:
+    // Called when target media file generation succeed.
+    virtual void OnMediaFileReady(base::Value&& playlist_value,
```

Comment:
you don't really need Rvalue references to move data to this function. Just `base::Value playlist_value` is fine